carla/recourse_methods/catalog/cem/model.py

```diff
@@ -61,7 +61,21 @@ def __init__(self, sess, catalog_model: MLModel, hyperparams):
                 ae_params["d"],
             ],
         )
-        self.AE = ae.load(input_shape=len(catalog_model.feature_input_order))
+        if ae_params["train_ae"]:
+            self.AE = train_autoencoder(
+                ae,
+                self._mlmodel.data,
+                self._mlmodel.scaler,
+                self._mlmodel.encoder,
+                self._mlmodel.feature_input_order,
+                epochs=ae_params["epochs"],
+                save=True,
+            )
+        else:
+            try:
+                self.AE = ae.load(input_shape=len(catalog_model.feature_input_order))
+            except Exception as exc:
```

Comment:
excepting `Exception` is generally considered bad practice, if you do try/except you mostly want to catch a specific exception.

Reply:
👍